endpoints/auction_test.go

```diff
@@ -376,32 +378,134 @@ func TestCacheVideoOnly(t *testing.T) {
 }
 
 func TestShouldUsersync(t *testing.T) {
-	doTest := func(gdprApplies string, consent string, allowBidderSync bool, allowHostCookies bool, expectAllow bool) {
-		t.Helper()
+	tests := []struct {
+		description      string
+		signal           string
+		allowHostCookies bool
+		allowBidderSync  bool
+		wantAllow        bool
+	}{
+		{
+			description:      "Don't sync - GDPR off, host cookies allows and bidder sync disallows",
+			signal:           "0",
+			allowHostCookies: true,
+			allowBidderSync:  false,
+			wantAllow:        false,
+		},
```

Comment:
If the gdpr signal is no (don't enforce) and the host allows cookies, shouldn't we allow syncing? The bidder sync wouldn't apply since gdpr is not enforced?

This might be a test setup error since we're using s mock. The real `BidderSyncAllowed` would return true if the signal is no.

Reply:
Do you think I should just remove the cases that are unrealistic in this test given the internals of `HostCookiesAllowed` and `BidderSyncAllowed`? My thought here was to exercise the conditionals in the `shouldUserSync` method ignoring the real implementation of those methods by using the mocks that were already in place.

Reply:
@SyntaxNode mentioned offline that he thinks we should remove the unrealistic cases.

Reply:
We shouldn't test conditions that only the mock would pass (test relies on the mock behavior being different from the real behavior). In these cases we should check if the real behavior is already being tested for in the test for the object being mocked. If so we should get rid of the test here. If not, we should consider either adding a test on the original object or improving the mock here so that the code paths can be properly tested.

Reply:
That makes sense. I'm remembering now that this is testing the legacy auction endpoint and this will all be deleted soon so I don't think it's worth our time to improve the mock, plus the original object has pretty good coverage already with a story in my queue to further improve it. I removed a bunch of the unrealistic test cases and just focused on testing how `HostCookiesAllowed` and `BidderSyncAllowed` affect the return value.